app/src/main/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreference.kt

```diff
@@ -76,14 +76,20 @@ class RadioSearchEngineListPreference @JvmOverloads constructor(
             ViewGroup.LayoutParams.WRAP_CONTENT,
         )
 
+        val isLastGeneralOrCustomSearchEngine = state.searchEngines.filter {
+            it.isGeneral ||
+                it.type == SearchEngine.Type.CUSTOM
```

Comment:
Here and below: The `isGeneral` check should already account for the `CUSTOM` type of a search engine - https://github.com/mozilla-mobile/firefox-android/blob/f99a255860a9faa4a9845f206a6a7b4694871bc6/android-components/components/feature/search/src/main/java/mozilla/components/feature/search/storage/SearchEngineReader.kt#L74-L78